### ISSUE TITLE
Allow language parameter to be passed to read_localization

### ISSUE THREAD

**Issue by codegain** [roundcube/roundcubemail#7598]
Hi,

for my plugin to work, I had to do this to get the localization texts in a specific language from a skin:

```
$lang = 'en_US'; // Some language from $_GET
$oldLang = $_SESSION['language'];
$_SESSION['language'] = $lang;
$texts = $rcmail->read_localization(__DIR__.'/../../skins/'.$rcmail->config->system_skin.'/localization');
$_SESSION['language'] = $oldLang;
```

This is because there is no way to pass a `$lang` parameter to the function "read_localization":

```
    /**
     * Read localized texts from an additional location (plugins, skins).
     * Then you can use the result as 2nd arg to load_language().
     *
     * @param string $dir Directory to search in
     *
     * @return array Localization texts
     */
    public function read_localization($dir)
    {
        $lang   = $_SESSION['language'];
        $langs  = array_unique(array('en_US', $lang));
        $locdir = slashify($dir);
        $texts  = array();
        ....
```

Would it be possible to just add a `$lang` parameter and use it if it's set (Otherwise fallback to `$_SESSION`)?

**Comment by alecpl** [comment]
Why would you do this? I guess we could accept such a change. Create a pull request.

**Comment by codegain** [comment]
Because I want to switch the language of Roundcube based on a `$_GET` parameter. The plugin "listens" for the `startup` hook and loads the language via `$rcmail->load_language($lang)`. Unfortunately this does not load the localizations from the skin which are located at `skins/skinname/localizations` and defined in the `meta.json` file as added in #5853.

I didn't find any other way as to call `read_localization` manually and pass the texts to the `load_language` method. This however requires me to set `$_SESSION['language']` before calling `read_localization` and resetting it before calling `load_language`. `load_language` will only load the texts if the given language (f.e. "en_US") is different from the session's language (which would also be now "en_US").


TLDR:
```
// Users language is set to "de_DE" and he wants to switch to "en_US" ($_SESSION['language'] == 'de_DE')
$lang = 'en_US';

$_SESSION['language'] = $lang;
$texts = $rcmail->read_localization(__DIR__.'/../../skins/'.$rcmail->config->system_skin.'/localization'); // => loads texts for "en_US"
$rcmail->load_language($lang, $texts); // => doesn't load any texts because $lang == $_SESSION['language']
```

**Comment by alecpl** [comment]
I think you could use:
```
$lang = 'en_US';
$rcmail->load_language($lang);
$_SESSION['language'] = $lang;
$texts = $rcmail->read_localization(__DIR__.'/../../skins/'.$rcmail->config->system_skin.'/localization');
$rcmail->load_language($lang, [], $texts);
```
but I merged the PR anyway.
